src/cplscheme/CompositionalCouplingScheme.cpp

```diff
@@ -16,161 +18,116 @@ void CompositionalCouplingScheme::addCouplingScheme(
     const PtrCouplingScheme &couplingScheme)
 {
   PRECICE_TRACE();
-  _couplingSchemes.emplace_back(couplingScheme);
+
+  if (!couplingScheme->isImplicitCouplingScheme()) {
+    _explicitSchemes.emplace_back(couplingScheme);
+    return;
+  }
+
+  PRECICE_CHECK(_implicitScheme == nullptr,
+                "You attempted to define a second implicit coupling-scheme for a participant, which is not allowed. "
+                "Please use a multi coupling-scheme for true implicit coupling of multiple participants.");
```

Comment:
Can we work with xml tags here and print the name of the participant? And could be nicer to move this to the configuration class.

Reply:
Moving this to the configuration class gives us access to the Participant's name, so that would help.